internal/transport/http2_client.go

```diff
@@ -137,11 +137,23 @@ type http2Client struct {
 	connectionID uint64
 }
 
-func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr string) (net.Conn, error) {
-	if fn != nil {
-		return fn(ctx, addr)
+func dial(ctx context.Context, fn func(context.Context, string) (net.Conn, error), addr resolver.Address, useProxy bool, grpcUA string) (net.Conn, error) {
+	if fn == nil {
+		fn = func(fCtx context.Context, fAddr string) (net.Conn, error) {
+			if networkType := addr.Attributes.Value("network_type"); networkType != nil {
+				if networkTypeStr, ok := networkType.(string); ok {
+					return (&net.Dialer{}).DialContext(ctx, networkTypeStr, addr.Addr)
+				}
+				return nil, fmt.Errorf("network_type %v not of type string", networkType)
+			}
+			network, fAddr := parseDialTarget(fAddr)
```

Comment:
Do we have a test to cover `passthrough:///unix:///path`?

Reply:
We did, but in the recent change we got rid of parseDialTarget.